android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/performance/SyncPerformanceIntTestsHarness.kt

```diff
@@ -125,74 +124,151 @@ class SyncPerformanceIntTestsHarness : BaseStitchAndroidIntTest() {
             .getCollection(stitchOutputCollName)
     }
 
-    fun runPerformanceTestWithParams(
+    fun handleTestResult(
         testParams: TestParams,
+        runResult: RunResult,
+        resultId: ObjectId,
+        numDoc: Int,
+        docSize: Int
+    ): Boolean {
+        val failuresBson = runResult.failures.map { it.asBson }
+        if (failuresBson.size >= (testParams.numIters + 1) / 2) {
+            if (testParams.outputToStdOut) {
+                Log.d("perfTests",
+                    String.format("(FAILED) %s failed %d times (docSize=%d, numDocs=%d",
+                        testParams.testName, failuresBson.size, docSize, numDoc
+                    )
+                )
+                failuresBson.forEach { Log.d("perfTests", it.toJson()) }
+            }
+
+            if (testParams.outputToStitch) {
+                val filterDocument = Document("_id", resultId)
+                val updateDocument = Document(mapOf(
+                    "\$push" to Document("results", Document(mapOf(
+                        "numDocs" to numDoc,
+                        "docSize" to docSize,
+                        "success" to false,
+                        "failures" to failuresBson)))
+                ))
+                Log.d("perfTests", "Adding failures to stitch")
+                Tasks.await(outputColl.updateOne(filterDocument, updateDocument))
+            }
+
+            return false
+        } else {
+            if (testParams.outputToStdOut) {
+                Log.d("perfTests",
+                    String.format("(SUCCESS): %s (docSize: %d, numDocs=%d): %s",
+                        testParams.testName, docSize, numDoc, runResult.asBson.toJson()
+                    )
+                )
+            }
+
+            if (testParams.outputToStitch) {
+                val filterDocument = Document("_id", resultId)
+                var runResultBson = runResult.asBson
+                if (failuresBson.size > 0) {
+                    runResultBson.append("failures", failuresBson)
+                }
+                val updateDocument = Document("\$push", Document("results", runResultBson))
+                Tasks.await(outputColl.updateOne(filterDocument, updateDocument))
+            }
+
+            return true
+        }
+    }
+
+    private fun getPerformanceTestingContext(testParams: TestParams): SyncPerformanceTestContext {
+        if (testParams.stitchHostName == "https://stitch.mongodb.com") {
```

Comment:
[nit] extract this out to a constant somewhere called `PROD_STITCH_HOSTNAME`, and put it maybe in `SyncPerformanceTestUtils`